Not building on nix 25.11

- What I did: As instructed, I added
`sls-steam = {
url = "github:AceSLS/SLSsteam";
inputs.nixpkgs.follows = "nixpkgs";
    }; ` 
to my flake inputs. 

- What I intended: Building of SLS-steam

- What happened: Build failed.

Log: 

`In file included from src/update.cpp:1:
\> src/update.hpp:3:10: fatal error: curl/curl.h: No such file or directory
\>     3 | #include <curl/curl.h>
\>       |          ^~~~~~~~~~~~~
\> compilation terminated.
\> make: *** [Makefile:51: obj/update.o] Error 1
`

Steps taken: this seemed like a clear case of curl not declared as a dependency, so I forked the project and added it to buildInput.mkDerivation. 

But then the build still failed, now with: 

`
\> sh tools/library-inject/build.sh
\> cp tools/library-inject/library-inject.so bin/library-inject.so
\> sh tools/ticket-grabber/build.sh
\> tools/ticket-grabber/build.sh: line 6: dotnet: command not found
\> make: *** [Makefile:45: tools/ticket-grabber/bin/Release/net9.0/linux-x64/publish/ticket-grabber] Error 127
`

I believe this is again a missing dependency, now because of dotnet, but I'm not sure how to fix it.

--- a/nix-modules/default.nix
+++ b/nix-modules/default.nix
@@ -3,36 +3,55 @@
   lib,
   stdenv,
   pkgs,
-}:
-pkgs.pkgsi686Linux.stdenv.mkDerivation {
-  pname = "SLSsteam";
-  version = "${rev}";
-  src = ../.;
-
-  nativeBuildInputs = with pkgs; [
-    pkg-config
-    makeWrapper
-  ];
-
-  buildInputs = with pkgs.pkgsi686Linux; [
-    openssl
-    which
-  ];
-
-  buildPhase = ''
-    make clean
-    make
-  '';
-
-  installPhase = ''
-    mkdir -p $out/
-    cp bin/SLSsteam.so $out/
-  '';
-
-  meta = {
-    description = "Steamclient Modification for Linux";
-    homepage = "https://github.com/AceSLS/SLSsteam";
-    license = lib.licenses.agpl3Only;
-    platforms = lib.platforms.linux;
+  buildDotnetModule,
+  dotnetCorePackages,
+}: let
+  ticketGrabber = import ./ticket-grabber.nix {
+    inherit rev buildDotnetModule dotnetCorePackages;
   };
-}
+in
+  pkgs.pkgsi686Linux.stdenv.mkDerivation {
+    pname = "SLSsteam";
+    version = "${rev}";
+    src = ../.;
+
+    nativeBuildInputs = with pkgs; [
+      pkg-config
+      makeWrapper
+    ];
+
+    buildInputs = with pkgs.pkgsi686Linux; [
+      openssl
+      which
+      curl
+    ];
+
+    buildPhase = ''
+      make clean
+
+      mkdir -p tools/ticket-grabber/bin/Release/net9.0/linux-x64/publish
+      cp ${ticketGrabber}/bin/ticket-grabber \
+       tools/ticket-grabber/bin/Release/net9.0/linux-x64/publish/ticket-grabber
+
+      make
+    '';
+
+    installPhase = ''
+      mkdir -p $out/
+      cp bin/SLSsteam.so $out/
+      cp bin/library-inject.so $out/
+
+      patchelf --set-rpath ${
+        lib.makeLibraryPath [
+          pkgs.pkgsi686Linux.curl
+        ]
+      } $out/SLSsteam.so
+    '';
+
+    meta = {
+      description = "Steamclient Modification for Linux";
+      homepage = "https://github.com/AceSLS/SLSsteam";
+      license = lib.licenses.agpl3Only;
+      platforms = lib.platforms.linux;
+    };
+  }

--- a/nix-modules/ticket-grabber.nix
+++ b/nix-modules/ticket-grabber.nix
@@ -1,0 +1,18 @@
+{
+  buildDotnetModule,
+  dotnetCorePackages,
+  rev,
+}: let
+  shortRev = builtins.substring 0 7 rev;
+in
+  buildDotnetModule {
+    pname = "Ticket-Grabber";
+    version = "0.1.0+${shortRev}";
+
+    src = ../tools/ticket-grabber;
+
+    projectFile = "ticket-grabber.csproj";
+    dotnet-sdk = dotnetCorePackages.sdk_9_0;
+    dotnet-runtime = dotnetCorePackages.runtime_9_0;
+    nugetDeps = ./deps.json;
+  }

--- a/nix-modules/wrapped.nix
+++ b/nix-modules/wrapped.nix
@@ -19,7 +19,7 @@ in
     installPhase = ''
       mkdir -p $out/bin
       makeWrapper ${pkgs.steam}/bin/steam $out/bin/SLSsteam \
-        --set LD_AUDIT "${sls-steam}/SLSsteam.so"
+        --set LD_AUDIT "${sls-steam}/library-inject.so:${sls-steam}/SLSsteam.so"
     '';
 
     meta = {

